src/session.js

```diff
@@ -0,0 +1,38 @@
+let crypto = require('crypto');
+
+let sessions = {};
+
+function hydrate(request, sessionId) {
+  let session = sessions[sessionId];
+  request.serial = session.serial;
+  request.remoteHost = session.remoteHost;
+  request.remotePort = session.remotePort;
+}
+
+function createHash(data) {
+  let hash = crypto.createHash('sha1');
```

Comment:
Yes, by specifying an HTTP header of `Session-Id` you will skip the crypto work and have your save serial, remote host and port session.